rom/dev/src/flow/cold_reset/idev_id.rs

```diff
@@ -264,37 +262,49 @@ impl InitDevIdLayer {
         cprintln!("[idev] SIG.S = {}", HexBytes(&_sig_s));
 
         // Build the CSR with `To Be Signed` & `Signature`
-        let mut csr = [0u8; MAX_CSR_SIZE];
+        let mut csr_buf = [0; MAX_CSR_SIZE];
         let result = Ecdsa384CsrBuilder::new(tbs.tbs(), &sig.to_ecdsa())
             .ok_or(CaliptraError::ROM_IDEVID_CSR_BUILDER_INIT_FAILURE);
         sig.zeroize();
 
         let csr_bldr = result?;
         let csr_len = csr_bldr
-            .build(&mut csr)
+            .build(&mut csr_buf)
             .ok_or(CaliptraError::ROM_IDEVID_CSR_BUILDER_BUILD_FAILURE)?;
 
-        if csr_len > csr.len() {
+        if csr_len > csr_buf.len() {
             return Err(CaliptraError::ROM_IDEVID_CSR_OVERFLOW);
         }
 
-        cprintln!("[idev] CSR = {}", HexBytes(&csr[..csr_len]));
+        cprintln!("[idev] CSR = {}", HexBytes(&csr_buf[..csr_len]));
         report_boot_status(IDevIdMakeCsrComplete.into());
 
+        let dev_id_csr = IDevIDCsr::new(&csr_buf, csr_len)?;
+
         // Execute Send CSR Flow
-        let result = Self::send_csr(env, InitDevIdCsr::new(&csr, csr_len));
-        csr.zeroize();
+        let mut result = Self::send_csr(env, &dev_id_csr);
+        if result.is_ok() {
+            result = Self::write_csr_to_peristent_storage(env, dev_id_csr);
+        }
+        csr_buf.zeroize();
 
         result
     }
 
+    fn write_csr_to_peristent_storage(env: &mut RomEnv, csr: IDevIDCsr) -> CaliptraResult<()> {
```

Comment:
nit: are we confident this won't make a copy on the stack?

Reply:
Changed to a Ref + Clone instead of the move.

We can follow up with a PR to assembly the CSR directly in the PersistentData memory slice if we want to save the stack space.